x/rollup/tx/helpers/l1_data_decorator.go

```diff
@@ -0,0 +1,91 @@
+package helpers
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/rlp"
+	rollupkeeper "github.com/polymerdao/monomer/x/rollup/keeper"
+)
+
+// ConsumeGasForL1DataDecorator will consume gas to compensate the sequencer
+// for posting the transaction to Ethereum. The gas cost is calculated based
+// on the Ecotone upgrade and the sequencer is expected to post the transaction
+// using blobs.
+type ConsumeGasForL1DataDecorator struct {
+	rollupKeeper *rollupkeeper.Keeper
+}
+
+func NewConsumeGasForL1DataDecorator(rollupKeeper *rollupkeeper.Keeper) ConsumeGasForL1DataDecorator {
+	return ConsumeGasForL1DataDecorator{
+		rollupKeeper: rollupKeeper,
+	}
+}
+
+func (d ConsumeGasForL1DataDecorator) AnteHandle(
+	ctx sdk.Context, //nolint:gocritic // hugeparam
+	tx sdk.Tx,
+	simulate bool,
+	next sdk.AnteHandler,
+) (sdk.Context, error) {
+	l1BlockInfo, err := d.rollupKeeper.GetL1BlockInfo(ctx)
```

Comment:
I looked into adding unit/integration test coverage for a while and eventually decided to split it out for the time being since it was blowing up in scope.

I created https://github.com/polymerdao/monomer/issues/281 to refactor how we pass around the rollup keeper to use an interface that can be mocked instead of a struct and to add unit tests for `ante.go` and `l1_data_decorator.go`.

Also, this logic will eventually be tested in our e2e tests once https://github.com/polymerdao/monomer/pull/267 merges.